code/modules/mob/living/silicon/ai/ai.dm

```diff
@@ -75,7 +75,7 @@
 	var/turf/waypoint //Holds the turf of the currently selected waypoint.
 	var/waypoint_mode = FALSE		//Waypoint mode is for selecting a turf via clicking.
 	var/call_bot_cooldown = 0		//time of next call bot command
-	var/apc_override = FALSE		//hack for letting the AI use its APC even when visionless
+	var/obj/machinery/power/apc/apc_override //Ref of the AI's APC, used when the AI has no power in order to access their APC.
```

Comment:
Is this reference properly cleared if the APC gets deleted?

Reply:
Yes.
`/mob/living/silicon/ai/proc/start_RestorePowerRoutine()`, `/mob/living/silicon/ai/proc/ai_restore_power()` and `/obj/machinery/power/apc/ui_state(mob/user)` all have handling for if a the local area APC is not detected.

Reply:
Oh, I see this was never a boolean to begin with, you just typed it properly